examples/ControlFlow.hs

```diff
@@ -34,9 +36,43 @@ import Control.Monad.Trans.Cont (ContT(..), callCC)
 import Streamly.Data.Stream (Stream)
 
 import qualified Streamly.Data.Fold as Fold
-import qualified Streamly.Data.Stream as Stream
+import qualified Streamly.Internal.Data.Stream as Stream
+import qualified Streamly.Internal.Data.Stream.StreamK.Type as K
 
+------------------------------------------------------------------------------
+-- Applicative/Monad instances
+------------------------------------------------------------------------------
+instance Monad m => Applicative (Stream m) where
 
+    {-# INLINE pure #-}
+    pure x = Stream.fromPure x
+
+    {-# INLINE (*>) #-}
+    s1 *> s2 = Stream.crossApplySnd s1 s2
+
+instance Monad m => Monad (Stream m) where
+    return = pure
+
+    {-# INLINE (>>=) #-}
+    (>>=) m f =
+            Stream.fromStreamK
+                $ K.bindWith
+                    K.serial
+                    (Stream.toStreamK m)
+                    (Stream.toStreamK . f)
+
+    {-# INLINE (>>) #-}
+    (>>) = (*>)
```

Comment:
We have a CrossStream type with these instances.

Reply:
We wanted to use the updated Stream.